src/DynamoCore/Library/TypedParameter.cs

```diff
@@ -1,42 +1,33 @@
 ﻿using System;
```

Comment:
None of the callers of `TypedParameter` constructor pass in `FunctionDescriptor`, they are always `null`. This pull request removes that redundant constructor. A new method `TypedParameter.UpdateFunctionDescriptor` is defined so that `FunctionDescriptor` can be updated after a `TypedParameter` object is constructed. During the call to `UpdateFunctionDescriptor` call, an optional `IPathManager` can be specified to help locating documentation assembly for the given `TypedParameter`.